client/core/core.go

```diff
@@ -1674,7 +1674,7 @@ func (c *Core) Run(ctx context.Context) {
 	defer c.walletMtx.Unlock()
 	for assetID, wallet := range c.wallets {
 		delete(c.wallets, assetID)
-		if !wallet.connected() {
+		if !wallet.connected() || wallet.parent != nil {
```

Comment:
If I'm reading https://github.com/decred/dcrdex/issues/2058#issuecomment-1399829244 correctly, we should actually be deleting this block and moving the `wallet.parent != nil` check to `Disconnect`.